BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -0,0 +1,1041 @@
+                    // Execute a burn to circularize our orbit at the current altitude.
+
+                    currentStatus = "Correcting Orbit (Circularizing)";
+
+                    Vector3d fvel, deltaV = Vector3d.up * 100;
+                    fc.throttle = 1;
+
+                    while (UnderTimeLimit() && deltaV.magnitude > 2)
+                    {
+                        yield return new WaitForFixedUpdate();
+
+                        UT = Planetarium.GetUniversalTime();
+                        fvel = Math.Sqrt(o.referenceBody.gravParameter / o.GetRadiusAtUT(UT)) * o.Horizontal(UT);
+                        deltaV = fvel - vessel.GetObtVelocity();
+
+                        fc.attitude = deltaV.normalized;
+                        fc.throttle = Mathf.Lerp(0, 1, (float)(deltaV.magnitude / 10));
+                    }
+                }
+            }
+            else if (targetVessel != null && hasWeapons)
+            {
+
+                // todo: implement for longer range movement.
+                // https://github.com/MuMech/MechJeb2/blob/dev/MechJeb2/MechJebModuleRendezvousAutopilot.cs
+                // https://github.com/MuMech/MechJeb2/blob/dev/MechJeb2/OrbitalManeuverCalculator.cs
+                // https://github.com/MuMech/MechJeb2/blob/dev/MechJeb2/MechJebLib/Maths/Gooding.cs
+
+                float minRange = Mathf.Max(MinEngagementRange, targetVessel.GetRadius() + vesselStandoffDistance);
+                float maxRange = Mathf.Max(weaponManager.gunRange, minRange * 1.2f);
+
+                float minRangeProjectile = minRange;
+                bool complete = false;
+                bool usingProjectile = true;
+
+                vessel.ActionGroups.SetGroup(KSPActionGroup.RCS, true);
+
+                if (weaponManager != null && weaponManager.selectedWeapon != null)
+                {
+                    currentWeapon = weaponManager.selectedWeapon;
+                    minRange = Mathf.Max((currentWeapon as EngageableWeapon).engageRangeMin, minRange);
+                    maxRange = Mathf.Min((currentWeapon as EngageableWeapon).engageRangeMax, maxRange);
+                    usingProjectile = weaponManager.selectedWeapon.GetWeaponClass() != WeaponClasses.Missile;
+                }
+
+                float currentRange = VesselDistance(vessel, targetVessel);
+                bool nearInt = false;
+                Vector3 relVel = RelVel(vessel, targetVessel);
+
+                if (currentRange < (!usingProjectile ? minRange : minRangeProjectile) && AwayCheck(minRange))
+                {
+                    currentStatus = "Maneuvering (Away)";
+                    fc.throttle = 1;
+                    fc.alignmentToleranceforBurn = 135;
+
+                    while (UnderTimeLimit() && targetVessel != null && !complete)
+                    {
+                        fc.attitude = FromTo(vessel, targetVessel).normalized * -1;
```

Comment:
`FromTo(targetVessel, vessel).normalized` would be more efficient than finding the vector then multiplying by `-1`.
Similarly, other vectors are being multiplied by `-1` further down, which would be better to simply use `+` or `-` as appropriate.